### DISCUSSION TITLE
[RFC] ConfigProvider.config support holderRender

### DISCUSSION THREAD

**Discussion by crazyair** [ant-design/ant-design#46615] [RFCs]
# 📋 RFC

## Summary

Static function `message.success(’xxx‘)` cannot set `hashPriority`, provide `container` support for customization

## Basic example

```tsx
ConfigProvider.config({
    container: children => (
        <StyleProvider hashPriority="high">
            <ConfigProvider prefixCls="test" modal={{ className: 'modal' }} message={{ className: 'message' }}>
                <App message={{ maxCount: 1 }}>{children}</App>
            </ConfigProvider>
        </StyleProvider>
    ),
});
```

## Drawbacks

* `container` no support customer element, can only use `Provider`

## Adoption strategy

Update in feature branch

**Comment by crazyair** [comment]
https://github.com/ant-design/ant-design/pull/46596

**Comment by crazyair** [comment]
`container` => `providerWrapper`?

**Comment by MadCcc** [comment]
`holderRender`

**Comment by crazyair** [comment]
https://github.com/ant-design/ant-design/pull/46596
